pkg/logs/input/docker/launcher.go

```diff
@@ -184,7 +184,9 @@ func (l *Launcher) overrideSource(container *Container, source *config.LogSource
 		return source
 	}
 
-	return newOverridenSource(standardService, shortName, source.Status)
+	newSource := newOverridenSource(standardService, shortName, source.Status)
```

Comment:
I'm slightly worried about this `parent` thing in this struct, especially since it is already a complex struct.
On top of that, calling it a Parent may be even more confusing, don't you think?

Here, can't we just copy (add) the `BytesRead` into the new one?

Reply:
I actually introduced this approach in [this PR](https://github.com/DataDog/datadog-agent/pull/6719) for a related feature. It's a bit tricky because at the time status is assembled, it only has context of the original (non overridden) sources. Since "collect all container logs" overrides the original source, there is no way for status to collect data on the child/overridden sources. Even if it could, in this scenario we want to only show a single source for all docker logs so we need to consolidate the data somehow. 

I am open for a better suggestion on this - as I will also need it for the other feature PR.